src/app/DIAmeterApplication.cpp

```diff
@@ -0,0 +1,558 @@
+
+  const string index = input_index;
+  string peptides_file = FileUtils::Join(index, "pepix");
+  string proteins_file = FileUtils::Join(index, "protix");
+  string auxlocs_file = FileUtils::Join(index, "auxlocs");
+
+  // params
+  double bin_width_  = Params::GetDouble("mz-bin-width");
+  double bin_offset_ = Params::GetDouble("mz-bin-offset");
+  vector<int> negative_isotope_errors = TideSearchApplication::getNegativeIsotopeErrors();
+  // bool use_neutral_loss_peaks_ = Params::GetBool("use-neutral-loss-peaks");
+  // bool use_flanking_peaks_ = Params::GetBool("use-flanking-peaks");
+
+  // Read proteins index file
+  ProteinVec proteins;
+  pb::Header protein_header;
+  if (!ReadRecordsToVector<pb::Protein, const pb::Protein>(&proteins, proteins_file, &protein_header)) { carp(CARP_FATAL, "Error reading index (%s)", proteins_file.c_str()); }
+
+  // Read auxlocs index file
+  vector<const pb::AuxLocation*> locations;
+  if (!ReadRecordsToVector<pb::AuxLocation>(&locations, auxlocs_file)) { carp(CARP_FATAL, "Error reading index (%s)", auxlocs_file.c_str()); }
+
+  // Read peptides index file
+  pb::Header peptides_header;
+  HeadedRecordReader* peptide_reader = new HeadedRecordReader(peptides_file, &peptides_header);
+  if ((peptides_header.file_type() != pb::Header::PEPTIDES) || !peptides_header.has_peptides_header()) { carp(CARP_FATAL, "Error reading index (%s)", peptides_file.c_str()); }
+
+  const pb::Header::PeptidesHeader& pepHeader = peptides_header.peptides_header();
+  DECOY_TYPE_T headerDecoyType = (DECOY_TYPE_T)pepHeader.decoys();
+  int decoysPerTarget = pepHeader.has_decoys_per_target() ? pepHeader.decoys_per_target() : 0;
+  carp(CARP_DEBUG, "decoysPerTarget = %d \t headerDecoyType=%d.", decoysPerTarget, headerDecoyType);
+
+  MassConstants::Init(&pepHeader.mods(), &pepHeader.nterm_mods(), &pepHeader.cterm_mods(), &pepHeader.nprotterm_mods(), &pepHeader.cprotterm_mods(), bin_width_, bin_offset_);
+  ModificationDefinition::ClearAll();
+  TideMatchSet::initModMap(pepHeader.mods(), ANY);
+  TideMatchSet::initModMap(pepHeader.nterm_mods(), PEPTIDE_N);
+  TideMatchSet::initModMap(pepHeader.cterm_mods(), PEPTIDE_C);
+  TideMatchSet::initModMap(pepHeader.nprotterm_mods(), PROTEIN_N);
+  TideMatchSet::initModMap(pepHeader.cprotterm_mods(), PROTEIN_C);
+
+
+  string output_file_name_ = make_file_path("tide-search.txt");
+  ofstream* output_file = create_stream_in_path(output_file_name_.c_str(), NULL, Params::GetBool("overwrite"));
+  TideMatchSet::writeHeadersDIA(output_file, Params::GetBool("compute-sp"));
+
+  map<string, double> peptide_predrt_map;
+  getPeptidePredRTMapping(&peptide_predrt_map);
+
+  /*
+  vector<InputFile> ms1_spectra_files = getInputFiles(input_files, 1);
+  vector<InputFile> ms2_spectra_files = getInputFiles(input_files, 2);
+
+
+  // Loop through spectrum files
+  for (pair<vector<InputFile>::const_iterator, vector<InputFile>::const_iterator> f(ms1_spectra_files.begin(), ms2_spectra_files.begin());
```

Comment:
This line and the next line are really long. Could you put them on multiple lines?